lib/storage.js

```diff
@@ -312,6 +312,20 @@ Storage.prototype.storeAddressAndWallet = function(wallet, addresses, cb) {
   });
 };
 
+Storage.prototype.fetchAddress = function(address, cb) {
+  var self = this;
+
+  this.db.collection(collections.ADDRESSES).findOne({
+    address: address,
```

Comment:
Are we using indexes for Address? `ensureIndex` (http://docs.mongodb.org/manual/reference/method/db.collection.ensureIndex/)